models/classes/LtiReturnResponse.php

```diff
@@ -94,13 +94,7 @@ public function send()
      * @return bool
      */
     protected function requiresRedirect() {
-        if ($this->exception instanceof LtiInvalidLaunchDataException
-            || $this->exception instanceof LtiVariableMissingException
-        ) {
-            return true;
-        }
-
-        return false;
+        return $this->exception instanceof LtiException;
```

Comment:
As we discussed it would be better to create specific exception for unsupported browser/OS and group client exceptions, for which we have to redirect customer